src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/CodeActionHelper.cs

```diff
@@ -47,7 +47,18 @@ public static class CodeActionHelper
             var diagnostics = semanticModel.GetDiagnostics();
             var location = GetTextSpan(request, sourceText);
 
-            var pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Contains(location)).ToImmutableArray();
+            // Try to find exact match
+            var pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Equals(location)).ToImmutableArray();
+            // No exact match found, try approximate match instead
+            if (!pointDiagnostics.Any())
+            {
+                var firstMatchingDiagnostic = diagnostics.FirstOrDefault(d => d.Location.SourceSpan.Contains(location));
+                // Try to find other matches with the same location as the first approximate match
+                if (firstMatchingDiagnostic != null)
+                {
+                    pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Equals(firstMatchingDiagnostic.Location.SourceSpan)).ToImmutableArray();
+                }
+            }
             if (pointDiagnostics.Any())
```

Comment:
Note that this is an `ImmutableArray<Diagnostic>` and not an `IEnumerable<Diagnostics>`. Since it's a value type, you could test for `pointDiagnostics.Length > 0` and `pointDiagnostics[0].Location.SourceSpan` below if you don't want to box the struct.